### ISSUE TITLE
UnknownError about a plan with no provided order

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13486]
When running the following query:
```cypher
WITH * ORDER BY null
RETURN 0 ORDER BY null, COUNT { RETURN 0 AS n0 UNION RETURN 0 AS n0 }
```

I encountered the following error:
```
Neo4jError: Neo.DatabaseError.General.UnknownError (While marking leveraged order we encountered a plan with no provided order. This is a bug.
.projection("NULL AS `  NULL@13`")
.apply()
.|.aggregation(Seq(), Seq("count(*) AS `  UNNAMED15`"))
.|.apply()
.|.|.argument()
.|.apply()
.|.|.distinct("`  n0@8` AS `  n0@8`")
.|.|.union()
.|.|.|.projection("`  n0@10` AS `  n0@8`")
.|.|.|.projection("$`  AUTOINT2` AS `  n0@10`")
.|.|.|.argument()
.|.|.projection("`  n0@9` AS `  n0@8`")
.|.|.projection("$`  AUTOINT1` AS `  n0@9`")
.|.|.argument()
.|.argument()
.apply()
.|.argument()
.sort("`  NULL@12` ASC")
.projection("NULL AS `  NULL@12`")
.argument()
.build())
```

I believe the query mentioned above is semantically and syntactically correct and thus no error should be thrown here.

I encountered this issue when testing queries against the **Neo4j 5.21 enterprise version** in a Docker container running **alpine v.3**.

### Steps to reproduce
Run the following query and observe it throws an error:
```cypher
WITH * ORDER BY null RETURN 0 ORDER BY null, COUNT { RETURN 0 AS n0 UNION RETURN 0 AS n0 }
```

### Expected behavior
The query should run successfully

### Actual behavior
The query fails with the error message `Neo4jError: Neo.DatabaseError.General.UnknownError (While marking leveraged order we encountered a plan with no provided order. This is a bug. [...]`

**Comment by SachinVasant** [comment]
Hi @DominicWuest, 
Thank you for reporting this issue. We will investigate it further and keep you posted.

**Comment by inmost-light** [comment]
@DominicWuest Should be fixed in 5.23.
